roles/generate-jenkins/templates/CONTRIBUTING.j2

```diff
@@ -0,0 +1,117 @@
+# Contributing to {{ project_name }}
+
+## Gotchas
+
+* While contributing make sure to make all your changes before creating a Pull Request, as our pipeline builds each commit after the PR is open.
+* Read, and fill the Pull Request template
+  * If this is a fix for a typo in code or documentation in the README please file an issue
+  * If the PR is addressing an existing issue include, closes #\<issue number>, in the body of the PR commit message
+
+## Common files
+
+| File | Use case |
+| :----: | --- |
+| `Dockerfile` | Dockerfile used to build amd64 images |
+| `Dockerfile.aarch64` | Dockerfile used to build 64bit ARM architectures |
+| `Dockerfile.armhf` | Dockerfile used to build 32bit ARM architectures |
+| `Jenkinsfile` | This file is a product of our builder and should not be edited directly. This is used to build the image |
+| `jenkins-vars.yml` | This file is used to generate the `Jenkinsfile` mentioned above, it only affects the build-process |
+| `package_versions.txt` | This file is generated as a part of the build-process, it lists all the installed packages and its version |
+| `README.md` | This file is a product of our builder and should not be edited directly. This displays the readme for the repository and image registries |
+| `readme-vars.yml` | This file is used to generate the `README.md` |
+
+## Readme
+
+If you would like to change our readme, please __**do not**__ directly edit the readme, as it is generated on each commit.
```

Comment:
Is it a decent idea to add a comment in the README.md file saying `DO NOT EDIT`? Bah, probably not. Either way I assume the odd one will slip through without reading!

Reply:
Markdown has no concepts of comments. This also updates the pr template to mention the contributing file

Reply:
Pretty sure you can do HTML comments actually

Reply:
I stand corrected then 😄